File: 2020/swift/Day01-2020.swift
import Foundation

func Day01() {
    let input = try! String(contentsOfFile: "day01.data")
    let numbers = input.split(separator: "\n").compactMap { Int($0) }
    
    Part1(numbers: numbers)
    Part2(numbers: numbers)
}

fileprivate func Part1(numbers: [Int]) {
    for x in numbers {
        for y in numbers {
            if x != y && x + y == 2020 {
                print(x * y)
                return
            }
        }
    }
}

fileprivate func Part2(numbers: [Int]) {
    for x in numbers {
        for y in numbers {
            for z in numbers {
                if x != y && x != z && y != z && x + y + z == 2020 {
                    print(x * y * z)
                    return
                }
            }
        }
    }
}

File: 2020/c#/Day01-2020.cs
using System;
using System.IO;
using System.Linq;
using System.Collections.Generic;

namespace AdventOfCode.Year2020
{
	public class Day01
	{
		public Day01()
		{
			var data = File.ReadAllLines("day01.data");
			var numbers = data.Select((x) => int.Parse(x));

			Part1(numbers);
			Part2(numbers);
		}

		void Part1(IEnumerable<int> numbers)
		{
			foreach (var x in numbers)
			{
				foreach (var y in numbers)
				{
					if (x != y && x + y == 2020)
					{
						Console.WriteLine($"{x * y}");
						return;
					}
				}
			}
		}

		void Part2(IEnumerable<int> numbers)
		{
			foreach (var x in numbers)
			{
				foreach (var y in numbers)
				{
					foreach (var z in numbers)
					{
						if (x != y && x != z && y != z && x + y + z == 2020)
						{
							Console.WriteLine($"{x * y * z}");
							return;
						}
					}
				}
			}
		}
	}
}
